util/src/length_prefix_encoding/encoder.rs

```diff
@@ -92,6 +92,90 @@ pub struct WriteToIoReturn {
 }
 
 #[derive(Clone, Copy, Debug)]
+/// Module description goes here (skipped since it already exists in #479)
+///
+/// # Examples
+///
+///  Unfinished/proof-of-concept versions below - don't try this at home
+///
+/// **TODOs**:
+///
+/// * Actually write to a sink, if that's desirable/integration tests don't suffice
+/// * Consider splitting into multiple examples in that case
+/// * Resolve all TBD notes, fixup/polish, add references/links, etc.
+/// * rustfmt ignores docstrings, check [unstable option](https://github.com/rust-lang/rustfmt/issues/3348) or fix manually?
+/// * Decide on the most important use cases vs. per-function examples (intended API usage?)
+/// * Unit test vs docstring test - unclear which approach is preferable here
+///
+/// ## Initialization and Consumption
+///
+/// The prefix-encoded message can be written as a whole or in parts:
+///
+/// ```
+/// use rosenpass_util::length_prefix_encoding::encoder::LengthPrefixEncoder;
+///
+/// let msg = String::from("hello world");
+/// let encoder = LengthPrefixEncoder::from_message(msg.as_bytes());
+/// assert!(encoder.encoded_message_bytes() > msg.len());
+/// assert!(!encoder.exhausted());
+///
+/// // TBD: should demonstrate several ways of creating new encoders?
+/// // TBD: relevant use cases to cover? need to mention trade-offs/perf/sec/other concerns?
+///
+/// // TBD: write_all_to_stdio -> assert that the entire message has been written (?)
```

Comment:
How about just having a check in the integration test above ensuring that the right data was written.